### PR TITLE
test and coverage reporting to azure

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes nothing.

### CODE REVIEW

**Comment by nabobalis** [comment]
The bot didn't appear?

**Comment by nabobalis** [comment]
It did appear if I did that.

**Comment by nabobalis** [comment]
Want me to also setup town crier?

**Comment by DanRyanIrish** [comment]
I'm going to defer to your and @Cadair's recommendation about towncrier.

**Comment by nabobalis** [comment]
I'm not fussed either way. It is your package!

**Comment by DanRyanIrish** [comment]
But I don't know what it'll do.  In total ignorance I would think that if towncrier is what sunpy does, then perhaps ndcube should do it too.  Unless @Cadair believes it's overkill in some way for ndcube.

**Comment by nabobalis** [comment]
Yes it would be the same method as the main package. Probably have to port over the changes to here to avoid having a hard dependency on SunPy

**Comment by DanRyanIrish** [comment]
Yes, we want to restrict ndcube's sunpy dependency to the visualization mixin.  We are planning of making that an optional dependency.

**Comment by Cadair** [comment]
Having a sunpy dependency for an optional part of the doc build should be fine?

**Comment by DanRyanIrish** [comment]
How would that affect sunpy users and developers who don’t want sunpy?

**Comment by Cadair** [comment]
You wouldn't be able to see the changelog in the doc build (like this: https://docs.sunpy.org/en/latest/whatsnew/changelog.html ) unless you had sunpy installed.

**Comment by nabobalis** [comment]
That should (at least from what I recall) be the full setup now for towncrier.

File: ndcube/utils/towncrier.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+"""
```
Comment (Cadair) [inline ndcube/utils/towncrier.py]: I feel like we really can just import this from sunpy.

Comment (nabobalis) [reply ndcube/utils/towncrier.py]: #yourchoicebutwouldavoidaharddeponsunpyifyoujustwantobuildthedocs

Comment (Cadair) [reply ndcube/utils/towncrier.py]: just skip writing out the changelog if you can't import sunpy. We have a hard dep on sunpy anyway at the moment, we can worry about refactoring it later.

Comment (nabobalis) [reply ndcube/utils/towncrier.py]: Sorted.

Comment (Cadair) [reply ndcube/utils/towncrier.py]: Thanks!

**Comment by DanRyanIrish** [comment]
Thanks guys!

**Pull request merged**